packages/ifpack2/test/unit_tests/Ifpack2_UnitTestAdditiveSchwarz.cpp

```diff
@@ -1249,6 +1249,300 @@ TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL(Ifpack2AdditiveSchwarz, SGS_Overlap, Scalar, L
+  params.set ("schwarz: reordering list", zlist);
+#else
+  params.set ("schwarz: use reordering", false);
+#endif
+
+  out << "Setting AdditiveSchwarz's parameters" << endl;
+
+  TEST_NOTHROW(prec.setParameters(params));
+
+  out << "Testing domain and range Maps of AdditiveSchwarz" << endl;
+
+  // FIXME (mfh 26 Jul 2015) The domain and range Maps of the
+  // preconditioner don't have to be the same object; they only need
+  // to be the same in the sense of Tpetra::Map::isSameAs().
+  //
+  //trivial tests to insist that the preconditioner's domain/range maps are
+  //identically those of the matrix:
+  const map_type* mtx_dom_map_ptr = &*crsmatrix->getDomainMap();
+  const map_type* mtx_rng_map_ptr = &*crsmatrix->getRangeMap();
+  const map_type* prec_dom_map_ptr = &*prec.getDomainMap();
+  const map_type* prec_rng_map_ptr = &*prec.getRangeMap();
+  TEST_EQUALITY( prec_dom_map_ptr, mtx_dom_map_ptr );
+  TEST_EQUALITY( prec_rng_map_ptr, mtx_rng_map_ptr );
+
+  out << "Calling AdditiveSchwarz's initialize()" << endl;
+  prec.initialize();
+
+  out << "Calling AdditiveSchwarz's compute()" << endl;
+  prec.compute();
+
+  MV x (rowmap, 2), y (rowmap, 2), z (rowmap, 2);
+  x.putScalar (one);
+
+  out << "Applying AdditiveSchwarz to a multivector" << endl;
+  prec.apply(x, y);
+
+  // The solution should now be full of 1/2s
+  z.putScalar (one / two);
+
+  Teuchos::ArrayRCP<const Scalar> yview = y.get1dView();
+  Teuchos::ArrayRCP<const Scalar> zview = z.get1dView();
+
+  TEST_COMPARE_FLOATING_ARRAYS(yview, zview, 4*STS::eps ());
+  #endif
+}
+
+TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL(Ifpack2AdditiveSchwarz, FastILDL, Scalar, LocalOrdinal, GlobalOrdinal)
+{ 
+#ifdef HAVE_IFPACK2_SHYLUFASTILU
+  typedef LocalOrdinal LO;
+  typedef GlobalOrdinal GO;
+
+  typedef Tpetra::CrsMatrix<Scalar,LO,GO,Node> crs_matrix_type;
+  typedef Tpetra::Map<LO,GO,Node> map_type;
+  typedef Tpetra::MultiVector<Scalar,LO,GO,Node> MV;
+  typedef Tpetra::RowMatrix<Scalar,LO,GO,Node> row_matrix_type;
+  typedef Teuchos::ScalarTraits<Scalar> STS;
+  const Scalar one = STS::one ();
+  const Scalar two = STS::one () + STS::one ();
+
+  out << "Ifpack2::AdditiveSchwarz: FastILDL" << endl;
+  Teuchos::OSTab tab1 (out);
+
+  global_size_t num_rows_per_proc = 5;
+
+  out << "Creating row Map and CrsMatrix" << endl;
+
+  RCP<const map_type> rowmap =
+    tif_utest::create_tpetra_map<LO, GO, Node> (num_rows_per_proc);
+  RCP<const crs_matrix_type> crsmatrix =
+    tif_utest::create_test_matrix<Scalar, LO, GO, Node> (rowmap);
+
+  out << "Creating AdditiveSchwarz instance" << endl;
+
+  Ifpack2::AdditiveSchwarz<row_matrix_type> prec (crsmatrix);
+  Teuchos::ParameterList params, zlist;
+
+  out << "Filling in ParameterList for AdditiveSchwarz" << endl;
+
+  zlist.set ("order_method", "rcm");
+  zlist.set ("order_method_type", "local");
+  params.set ("inner preconditioner name", "FILDL");
+  {
+    Teuchos::ParameterList innerParams;
+    innerParams.set("sweeps", 1);
```

Comment:
That's probably because the problem matrix is only 5 rows. I just copied the test logic from the unit test "Test0", which does AdditiveSchwarz with ILUT. I'll make the problem bigger and then increase sweeps.